robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+                         nearest=nearest,
+                         attempts=attempts,
+                         search_timeout=search_timeout,
+                         look_range=look_range,
+                         look_steps=look_steps),
+                     transitions={
+                         'found': 'UPDATE_POSE',
+                         'failed': 'failed'
+                     })
+
+            self.add("UPDATE_POSE",
+                     states.UpdateDestEntityPoseWithSrcEntity(
+                         robot=robot,
+                         src_entity_designator=found_person_designator,
+                         dst_entity_designator=dst_entity_designator,
+                         dst_entity_type=dst_entity_type),
+                     transitions={
+                         'done': 'done',
+                         'failed': 'failed'
+                     })
+
+
+class _DecideNavigateState(smach.State):
+    """ Helper state to decide whether to use a NavigateToWaypoint or a NavigateToRoom state
+    """
+    def __init__(self, robot, waypoint_designator, room_designator):
+        """ Initialize method
+        :param robot: Robot API object
+        :param waypoint_designator: EdEntityDesignator that should resolve to a waypoint
+        :param room_designator: EdEntityDesignator that should resolve to the room in which the waypoint is located
+        """
+        smach.State.__init__(self, outcomes=["waypoint", "room", "none"])
+        self._robot = robot
+        self._waypoint_designator = waypoint_designator
+        self._room_designator = room_designator
+
+    def execute(self, ud):
+
+        # First: see if the waypoint exists
+        entity = self._waypoint_designator.resolve()
+        if entity:
+            return "waypoint"
+
+        # If not: try the room
+        entity = self._room_designator.resolve()
+        if entity:
+            return "room"
+
+        return "none"
+
+
+class FindPeopleInRoom(smach.StateMachine):
+    """ Uses NavigateToWaypoint or NavigateToRoom and subsequently tries to find people in that room
+    in that room.
+    """
+
+    def __init__(self, robot, room, found_people_designator,
+                 look_range=(-np.pi/2, np.pi/2),
+                 look_steps=8):
+        """ Constructor
+        :param robot: robot object
+        :param area: (str) if a waypoint "<area>_waypoint" is present in the world model, the robot will navigate
+        to this waypoint. Else, it will navigate to the room called "<area>"
+        :param name: (str) Name of the person to look for
+        :param discard_other_labels: (bool) Whether or not to discard faces based on label
+        :param found_person_designator: (Designator) A designator that will resolve to the found object
+        """
+        smach.StateMachine.__init__(self, outcomes=["found", "not_found"])
+
+        waypoint_designator = ds.EntityByIdDesignator(robot=robot, id=room + "_waypoint")
```

Comment:
This seems like a semi-hardcoded waypoint name. Why not allow the robot to drive to any waypoint?